### ISSUE TITLE
Link Down , Unable to use MoonGen.

### ISSUE THREAD

**Issue by prashantharaonr** [emmericp/MoonGen#271]
Hello,

I am trying to use MOONGen, I got the following error. I did updates of the driver (one NIC Intel X550T2 and second Intel X550T)and the OS is Debian buster with kernel 4.19.0-8-amd64.
Kindly help  how to proceed.

user@user:~/MoonGen$ sudo ./build/MoonGen examples/l3-load-latency.lua 0 1
[INFO]  Initializing DPDK. This will take a few seconds...
EAL: Detected 8 lcore(s)
EAL: Probing VFIO support...
EAL: PCI device 0000:01:00.0 on NUMA socket -1
EAL:   Invalid NUMA socket, default to 0
EAL:   probe driver: 8086:15d1 net_ixgbe
EAL: PCI device 0000:02:00.0 on NUMA socket -1
EAL:   Invalid NUMA socket, default to 0
EAL:   probe driver: 8086:1563 net_ixgbe
EAL: PCI device 0000:02:00.1 on NUMA socket -1
EAL:   Invalid NUMA socket, default to 0
EAL:   probe driver: 8086:1563 net_ixgbe
[INFO]  Found 3 usable devices:
   Device 0: 00:00:00:00:00:00 (Intel Corporation Ethernet Controller 10G X550T)
   Device 1: 11:11:11:11:11:11 (Intel Corporation Ethernet Controller 10G X550T)
   Device 2: 22:22:22:22:22:22 (Intel Corporation Ethernet Controller 10G X550T)
PMD: ixgbe_dev_link_status_print():  Port 0: Link Down
PMD: ixgbe_dev_link_status_print():  Port 1: Link Down
[INFO]  Waiting for devices to come up...
[ERROR] Device 0 (00:00:00:00:00:00) is DOWN: 0 MBit/s
[ERROR] Device 1 (11:11:11:11:11:11) is DOWN: 0 MBit/s
[INFO]  0 devices are up.
[WARN]  Failed to set rate limiter on queue [TxQueue: id=0, qid=0]: Invalid argument
[INFO]  Performing ARP lookup on 10.1.0.10
[INFO]  Performing ARP lookup on 10.1.0.10
[INFO]  ARP lookup failed, using default destination mac address
[INFO]  ARP lookup failed, using default destination mac address
[WARN]  Packet size 60 is smaller than minimum timestamp size 84. Timestamped packets will be larger than load packets.

Regards
Rao

**Comment by harshgondaliya** [comment]
I think your links are not up. The following is what I am able to think of:

1. Please make sure that the entire link is UP. Meaning:

- Two ports to be binded to MoonGen are UP.

- Two Moongen ports are connected via working optic cable to two ports on the other side.

- (IMP) The two ports on the other end of link are also UP.

2. If you are ubuntu user try commands like:` ip link set eth0 up
`
On my ethernet card, there are LEDs that show whether the link is UP or DOWN. If such LEDs are available, check for its status.

**Comment by emmericp** [comment]
Does it work with Linux? Does it work with other DPDK examples?

**Comment by harshgondaliya** [comment]
Yes, I am using a Linux machine with Intel Ethernet Server Adapter X520-DA2 NIC. 
Everything works as expected for me.
But yes, while running l3-load-latency.lua, if ports on the other side of the link are down, then I too get the same error as _prashantharaonr_.

**Comment by jabir366** [comment]
Have you tried connecting a cable from port zero to port 1 ? this should work anyway,
